src/lib/cimb_translator/CimbDecoder.cpp

```diff
@@ -65,18 +65,23 @@ CimbDecoder::CimbDecoder(unsigned symbol_bits, unsigned color_bits, unsigned col
 	load_tiles();
 }
 
-const color_correction& CimbDecoder::get_ccm() const
+// protected
+color_correction& CimbDecoder::internal_ccm() const
 {
-	// testing purposes only.
-	// this returning a thread local would be fine, iff we only use it for debugging!
+	static thread_local color_correction _ccm;
 	return _ccm;
```

Comment:
The only interesting code change on this branch.

We've had a subtle(?) race condition in the CFC app ever since I introduced color correction -- this `CimbDecoder` object is shared across threads, which means that updating the ccm matrix is asking for trouble.

I could pass it around as a function param (implicitly threadlocal), but I want to experiment with static threadlocal a bit -- it might be useful for the `Config` object...